app/presenters/licences/view-licence-purposes.presenter.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+/**
+ * Formats the licence purposes data for the view licence purposes page
+ * @module ViewLicencePurposesPresenter
+ */
+
+const { formatAbstractionPeriod } = require('../base.presenter.js')
+
+/**
+ * Formats the licence purposes data for the view licence purposes page
+ *
+ * @param {module:LicenceModel} licence - The licence and related licenceVersionPurposes data returned by
+ * `FetchLicencePurposesService`
+ *
+ * @returns {object} licence and licenceVersionPurposes data needed by the view template
+ */
+function go (licence) {
+  return {
+    id: licence.id,
+    licenceRef: licence.licenceRef,
+    licencePurposes: _formatLicencePurposes(licence.licenceVersions[0].licenceVersionPurposes),
+    pageTitle: 'Licence purpose details'
+  }
+}
+
+// NOTE: Maps over the `licenceVersionPurposes` array, formatting each `licenceVersionPurpose` object to extract
+// relevant information while ensuring uniqueness. A Set is used to track already processed licence details by
+// "stringify-ing" each object. If a duplicate is found, it returns null for that entry. After mapping, it filters out
+// any null values.
+function _formatLicencePurposes (licenceVersionPurposes) {
+  const uniquePurposeDetails = new Set()
+
+  return licenceVersionPurposes.map((licenceVersionPurpose) => {
+    const licenceString = JSON.stringify(licenceVersionPurpose)
+
+    if (uniquePurposeDetails.has(licenceString)) {
+      return null
+    } else {
+      uniquePurposeDetails.add(licenceString)
+    }
+
+    return {
+      abstractionAmounts: _formatAbstractionAmounts(licenceVersionPurpose),
+      abstractionPeriod: _abstractionPeriod(licenceVersionPurpose),
+      abstractionPoints: _formatAbstractionPoints(licenceVersionPurpose.points),
+      purposeDescription: licenceVersionPurpose.purpose.description ? licenceVersionPurpose.purpose.description : ''
```

Comment:
Can probably remove the check if exists since it is a set value in the purpose table so is never empty!